components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -765,9 +764,28 @@ public Monetization getMonetizationImplClass() throws APIManagementException {
         return monetizationImpl;
     }
 
+    private String getSubscriptionVersionRange(String versionRange, String apiVersion, String apiUUID) throws APIManagementException {
+        String subscriptionVersionRange = null;
+        if (versionRange != null) {
+            if(apiVersion.startsWith("v")) {
+                SemVersion apiSemVersion = SemanticVersionUtil.validateAndGetVersionComponents(apiVersion, apiUUID);
+                int apiMajorVersion = apiSemVersion.getMajor();
+                if (versionRange.equals("v" + apiMajorVersion)) {
+                    subscriptionVersionRange = versionRange;
+                } else {
+                    throw new APIManagementException(ExceptionCodes.from(ExceptionCodes.INVALID_SUBSCRIPTION_VERSION_RANGE,
+                            versionRange, apiVersion));
+                }
+            } else {
+                throw new APIManagementException(ExceptionCodes.from(ExceptionCodes.VERSION_RANGE_NOT_ALLOWED));
+            }
+        }
+        return subscriptionVersionRange;
+    }
+
     @Override
-    public SubscriptionResponse addSubscription(ApiTypeWrapper apiTypeWrapper, String userId, Application application)
```

Comment:
Let's not pass the null values to method arguments when calling from outside. Its very hard to read whether nulls are accepted. We can have an overidden method which doesn't have the nullable argument and call the 2nd method with passing null.